### ISSUE TITLE
DAGCircuitError with qiskit version 0.4.13 when running with ibmqx4 and ibmqx5

### ISSUE THREAD

**Issue by rraymondhp** [QISKit/qiskit-sdk-py#416]
When running the third cell of the iterative phase estimation tutorial, as below,
https://github.com/QISKit/qiskit-tutorial/blob/master/reference/algorithms/iterative_phase_estimation_algorithm.ipynb

```
Q_program = QuantumProgram()
Q_program.set_api(qx_config['APItoken'], qx_config['url']) # set the APIToken and API url

circuitName = "IPEAonReal_Step1"
#we then prepare quantum and classical registers and the circuit
qr = Q_program.create_quantum_register("qr", 5)
cr = Q_program.create_classical_register("cr", 5)
realStep1Circuit = Q_program.create_circuit(circuitName, [qr], [cr])

#apply IPEA
realStep1Circuit.h(qr[0])
for i in range(8):
    cu5pi8(realStep1Circuit, qr[0], qr[1])
realStep1Circuit.h(qr[0])
realStep1Circuit.measure(qr[0], cr[0])

backend = "ibmqx4"
backend_config = Q_program.get_backend_configuration(backend) #we get the config of backend to get its coupling map
shots = 1000
results = Q_program.execute([circuitName], backend=backend, shots=shots, coupling_map=backend_config["coupling_map"])
plot_histogram(results.get_counts(circuitName))
```

I got the following error:
```
---------------------------------------------------------------------------
DAGCircuitError                           Traceback (most recent call last)
<ipython-input-5-7d9d8902dd53> in <module>()
     19 shots = 1000
     20 results = Q_program.execute([circuitName], backend=backend, shots=shots, 
---> 21                             coupling_map=backend_config["coupling_map"], timeout=True)
     22 plot_histogram(results.get_counts(circuitName))

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/_quantumprogram.py in execute(self, name_of_circuits, backend, config, wait, timeout, basis_gates, coupling_map, initial_layout, shots, max_credits, seed, hpc)
   1519                             coupling_map=coupling_map, initial_layout=initial_layout,
   1520                             shots=shots, max_credits=max_credits, seed=seed,
-> 1521                             hpc=hpc)
   1522         result = self.run(qobj, wait=wait, timeout=timeout)
   1523         return result

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/_quantumprogram.py in compile(self, name_of_circuits, backend, config, basis_gates, coupling_map, initial_layout, shots, max_credits, seed, qobj_id, hpc)
   1171                 coupling_map=coupling_map,
   1172                 initial_layout=initial_layout,
-> 1173                 get_layout=True)
   1174             # making the job to be added to qobj
   1175             job = {}

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/_openquantumcompiler.py in compile(quantum_circuit, basis_gates, coupling_map, initial_layout, get_layout, format)
     78 
     79     dag_unroller = DagUnroller(compiled_dag_circuit, DAGBackend(basis))
---> 80     compiled_dag_circuit = dag_unroller.expand_gates()
     81     final_layout = None
     82     # if a coupling map is given compile to the map

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_dagunroller.py in expand_gates(self, basis)
    100                                                            current_node["params"],
    101                                                            current_node["qargs"],
--> 102                                                            current_node["condition"])
    103                 self.dag_circuit.substitute_circuit_one(node, subcircuit, wires)
    104         return self.dag_circuit

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_dagunroller.py in _build_subcircuit(self, gatedefs, basis, gate_name, gate_params, gate_args, gate_condition)
    164         sub_ast = Program(gatedefs + reg_nodes + [gate_node])
    165         # Interpret the AST to give a new DAGCircuit over backend basis
--> 166         sub_circuit = Unroller(sub_ast, DAGBackend(basis)).execute()
    167         return sub_circuit, new_wires
    168 

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_unroller.py in execute(self)
    278         """Interpret OPENQASM and make appropriate backend calls."""
    279         if self.backend is not None:
--> 280             self._process_node(self.ast)
    281             return self.backend.get_output()
    282         else:

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_unroller.py in _process_node(self, node)
    185         """Carry out the action associated with node n."""
    186         if node.type == "program":
--> 187             self._process_children(node)
    188 
    189         elif node.type == "qreg":

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_unroller.py in _process_children(self, node)
    180         """Call process_node for all children of node."""
    181         for kid in node.children:
--> 182             self._process_node(kid)
    183 
    184     def _process_node(self, node):

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_unroller.py in _process_node(self, node)
    220 
    221         elif node.type == "custom_unitary":
--> 222             self._process_custom_unitary(node)
    223 
    224         elif node.type == "universal_unitary":

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_unroller.py in _process_custom_unitary(self, node)
    110                                         self.arg_stack[0:-1])
    111                 if not self.gates[name]["opaque"]:
--> 112                     self._process_children(gbody)
    113                 self.backend.end_gate(name,
    114                                       [self.arg_stack[-1][s] for s in gargs],

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_unroller.py in _process_children(self, node)
    180         """Call process_node for all children of node."""
    181         for kid in node.children:
--> 182             self._process_node(kid)
    183 
    184     def _process_node(self, node):

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_unroller.py in _process_node(self, node)
    220 
    221         elif node.type == "custom_unitary":
--> 222             self._process_custom_unitary(node)
    223 
    224         elif node.type == "universal_unitary":

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_unroller.py in _process_custom_unitary(self, node)
    110                                         self.arg_stack[0:-1])
    111                 if not self.gates[name]["opaque"]:
--> 112                     self._process_children(gbody)
    113                 self.backend.end_gate(name,
    114                                       [self.arg_stack[-1][s] for s in gargs],

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_unroller.py in _process_children(self, node)
    180         """Call process_node for all children of node."""
    181         for kid in node.children:
--> 182             self._process_node(kid)
    183 
    184     def _process_node(self, node):

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_unroller.py in _process_node(self, node)
    226             qid = self._process_bit_id(node.children[1])
    227             for element in qid:
--> 228                 self.backend.u(args, element, self.arg_stack)
    229 
    230         elif node.type == "cnot":

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/unroll/_dagbackend.py in u(self, arg, qubit, nested_scope)
    105             self.circuit.apply_operation_back(
    106                 "U", [qubit], [], list(map(lambda x: x.sym(nested_scope),
--> 107                                            arg)), condition)
    108 
    109     def cx(self, qubit0, qubit1):

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/dagcircuit/_dagcircuit.py in apply_operation_back(self, name, qargs, cargs, params, condition)
    359         all_cbits.extend(cargs)
    360 
--> 361         self._check_basis_data(name, qargs, cargs, params)
    362         self._check_condition(name, condition)
    363         self._check_bits(qargs, self.output_map, False)

~/miniconda3/envs/QISKitenv/lib/python3.6/site-packages/qiskit/dagcircuit/_dagcircuit.py in _check_basis_data(self, name, qargs, cargs, params)
    261         if name not in self.basis:
    262             raise DAGCircuitError("%s is not in the list of basis operations"
--> 263                                   % name)
    264 
    265         # Check the number of arguments matches the signature

DAGCircuitError: 'U is not in the list of basis operations'

```

## Possible Solution

I have no idea. 
## Steps to Reproduce (for bugs)

1. Run the tutorial with the latest qiskit
2.
3.
4.

## Your Environment

* Version used: qiskit 0.4.13
* Environment name and version (e.g. Python 3.6.1): Python 3.6
* Operating System and version: MacOSX

**Comment by t-imamichi** [comment]
I make a minimal sample to reproduce this error.
https://gist.github.com/t-imamichi/b94d0e9a2339239ecd757f8d91b8bd70

**Comment by ajavadia** [comment]
fixed via #418
